src/react/src/components/Filters/NestedSelect.jsx

```diff
@@ -0,0 +1,232 @@
+    const { expandedGroups } = selectProps;
+    const isVisible = expandedGroups.some(group => data.groupLabel === group);
+
+    return isVisible ? <components.Option {...props} /> : null;
+};
+
+const DropdownIndicator = ({ selectProps: { classes } }) => (
+    <div className={classes.dropdownIndicator}>
+        <ArrowDropDownIcon />
+    </div>
+);
+
+const NestedSelect = ({
+    name,
+    label,
+    optionsData,
+    sectors,
+    updateSector,
+    classes,
+    isSideBarSearch,
+    windowWidth,
+    ...rest
+}) => {
+    const selectFilterStyles = makeSelectFilterStyles(windowWidth);
+    const nestedSelectFilterStyles = makeNestedSelectFilterStyles;
+    const combinedStyles = {
+        ...selectFilterStyles,
+        ...nestedSelectFilterStyles,
+    };
+    const [menuIsOpen, setMenuIsOpen] = useState(false);
+    const [expandedGroups, setExpandedGroups] = useState([]);
+
+    const handleInputChange = inputValue => {
+        if (inputValue.trim() === '') {
+            setExpandedGroups([]);
+            return;
+        }
+
+        const newExpandedGroups = optionsData
+            .filter(group =>
+                group.options.some(option =>
+                    option.label
+                        .toLowerCase()
+                        .includes(inputValue.toLowerCase()),
+                ),
+            )
+            .map(group => group.label);
+
+        setExpandedGroups(newExpandedGroups);
+    };
+
+    const handleSelect = selected => {
+        updateSector(selected);
+    };
+
+    const onMenuOpen = () => {
+        setMenuIsOpen(true);
+    };
+
+    const onMenuClose = () => {
+        setExpandedGroups([]);
+        setMenuIsOpen(false);
+    };
+
+    const customComponents = {
+        GroupHeading,
+        Option,
+        DropdownIndicator,
+        IndicatorSeparator: null,
+    };
+
+    const additionalProps = !isSideBarSearch
+        ? {
+              menuPosition: 'fixed',
+              menuPortalTarget: document.body,
+              closeMenuOnScroll: e =>
+                  menuIsOpen &&
+                  (e.target.classList === undefined ||
+                      !e.target.classList.contains('select__menu-list')),
+          }
+        : {};
+
+    return (
+        <>
+            <InputLabel
+                shrink={false}
+                htmlFor={name}
+                className={classes.inputLabelStyle}
+            >
+                {label}
+            </InputLabel>
+
+            <ReactSelect
+                isMulti
+                id={name}
+                name={name}
+                label={label}
+                options={optionsData}
+                value={sectors}
+                components={customComponents}
+                placeholder="Select"
+                className={`basic-multi-select notranslate ${classes.selectStyle}`}
+                classNamePrefix="select"
+                classes={classes}
+                styles={combinedStyles}
```

Comment:
There are `classes`, `styles` and `className` applied to the same component. Is it possible to group them into a single `className`?

Reply:
Ack.